src/pages/staticPages/tools/oncoprinter/OncoprinterClinicalAndHeatmapUtils.ts

```diff
@@ -503,14 +521,28 @@ export function getClinicalTracks(
                     attr.countsCategories!.length ===
                         MUTATION_SPECTRUM_FILLS.length
                 ) {
-                    countsCategoryFills = MUTATION_SPECTRUM_FILLS;
+                    countsCategoryFills = MUTATION_SPECTRUM_FILLS.slice();
+                    _.forEach(attr.countsCategories, (label, i) => {
```

Comment:
i think this code would be clearer if you used map and conditional inside of the map wherein you return either the default (from MUTATION_SPECTRUM_FILLS) or userSelectedClinicalTracksColors entry.    That brings the fundamental operation to the top (building countsCategoryFills)